h/api/models/document.py

```diff
@@ -40,7 +44,7 @@ def title(self):
     @classmethod
     def find_by_uris(cls, session, uris):
         """Find documents by a list of uris."""
-        query_uris = [text_type(uri.normalize(u), 'utf-8') for u in uris]
+        query_uris = [normalize(u) for u in uris]
```

Comment:
This might be the right time to look at making `uri.normalize` return a unicode string.